blog.tsx

```diff
@@ -291,6 +291,12 @@ export async function handler(
         "twitter:image": blogState.ogImage ?? blogState.cover,
         "twitter:card": blogState.ogImage ? "summary_large_image" : undefined,
       },
+      links: blogState.canonicalUrl ? [
+        {
+          rel: "canonical",
+          href: blogState.canonicalUrl,
+        }
+      ] : undefined,
```

Comment:
Why are these objects?

Reply:
That checks whether the canonical URL is set (in options) and if so, It adds a `<link rel="canonical" href="<!--url-->"/>`. Actually, I think it might be a good idea to first normalize the URL using `new URL(canonicalUrl).href`. Canonical tags are useful for SEO.